sdk/keyvault/keyvault/inc/az_keyvault.h

```diff
@@ -25,6 +25,15 @@ typedef enum {
   AZ_KEY_VAULT_KEY_TYPE_CERTIFICATE = 3,
 } az_key_vault_key_type;
 
+typedef enum {
+  AZ_KEY_VAULT_JSON_WEB_KEY_TYPE_NONE = 0,
+  AZ_KEY_VAULT_JSON_WEB_KEY_TYPE_EC = 1,
+  AZ_KEY_VAULT_JSON_WEB_KEY_TYPE_EC_HSM = 2,
+  AZ_KEY_VAULT_JSON_WEB_KEY_TYPE_RSA = 3,
+  AZ_KEY_VAULT_JSON_WEB_KEY_TYPE_RSA_HSM = 4,
+  AZ_KEY_VAULT_JSON_WEB_KEY_TYPE_OCT = 5,
+} az_keyvault_json_web_key_type;
```

Comment:
in the line above, you use `key_vault`, here you use `keyvault` - sometimes it is the first thing, sometimes is another, please be consistent in naming. I think it should be the underscored one, the `key_vault` - because the name of the service is "Key Vault".
Should you go with "key_vault", don't forget the file names and module names, such as `az_keyvault.h`

Reply:
Also, regarding the enum, can we get away with just having extern const az_spans, without defining the enum alltogether?

Reply:
Also, does `_JSON_WEB_` have to be the part of the name? Maybe it is, I don't know the domain enough; just a thing to consider.

Reply:
Big part of the SDK is a helper layer between JSON/Web REST interface of Azure and customer's code, is it worth mentioning? If Web is worth mentioning, is JSON worth to be mentioned too? Are there other possible `AZ_KEY_VAULT_x_WEB_KEY_TYPE`s, where `x != "JSON"`? Or, `AZ_KEY_VAULT_JSON_y_KEY_TYPE`s, where `y != "WEB"`? If both JSON and WEB are needed, should they go as `JSON_WEB`, or `WEB_JSON`?

Reply:
We need to have a big conversation about enums. Maybe we can do it next week but after Thanksgiving is more likely.

Reply:
thank you @antkmsft, updating all the keyvault to key_vault is a mayor refactor. 
I am creating a new issue for that since we need to take care or variable names, file names and folder names.
I wouldn't like to make this PR grow that much with all of the updates from that renaming.
Will do it next after this better
https://github.com/Azure/azure-sdk-for-c/issues/225